### PR TITLE
Remove deprecated createOrFetchReport method

### PR DESCRIPTION
**Pull request by stitesExpensify**
### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/238801

### Tests

1. Create a new chat with a single user and make sure there are no errors
2. Create a new chat with multiple users and make sure there are no errors
3. Open an existing chat from the LeftHandNav and make sure the report opens correctly
4. Open the search panel, find an existing chat and choose it. Make sure the report opens correctly
5. Send an IOU and make sure there are no errors
6. Go to that IOU on another account and pay it off

- [x] Verify that no errors appear in the JS console

### QA Steps

1. Create a new chat with a single user and make sure there are no errors
2. Create a new chat with multiple users and make sure there are no errors
3. Open an existing chat from the LeftHandNav and make sure the report opens correctly
4. Open the search panel, find an existing chat and choose it. Make sure the report opens correctly
5. Send an IOU and make sure there are no errors
6. Go to that IOU on another account and pay it off

- [ ] Verify that no errors appear in the JS console

#### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

#### Web
https://user-images.githubusercontent.com/42391420/199836182-5a147669-d8f3-4815-8196-b0fad20f1aca.mp4

#### Mobile Web - Chrome

https://user-images.githubusercontent.com/42391420/199836315-2bb7cc44-cff4-4e3b-96f4-d8ecef62c24f.mp4

#### Mobile Web - Safari

https://user-images.githubusercontent.com/42391420/199836288-e944052b-86ac-4bca-a9ae-8b4c474766e2.mp4

#### Desktop
https://user-images.githubusercontent.com/42391420/199836254-c39e0a4f-7180-42b8-b303-a2524c6eeb41.mp4

#### iOS

https://user-images.githubusercontent.com/42391420/199836350-3e27befe-d01b-4e9a-a479-6b7dd184e364.mp4

#### Android

https://user-images.githubusercontent.com/42391420/199836374-2acb89a2-1da7-4f7a-9189-aed0078c5c67.mp4

### CODE REVIEW

**Review by roryabraham — COMMENTED** [review]
Heads up – there were a few extra changes from [this PR](https://github.com/Expensify/App/pull/12357/files) that we should include over here

File: src/libs/Navigation/AppNavigator/AuthScreens.js
Comment (stitesExpensify) [inline src/libs/Navigation/AppNavigator/AuthScreens.js:89]: This timer was being started but never stopped / recorded, because the stop for it was in fetchAllReports, which was no longer being called anywhere.

**Comment by stitesExpensify** [comment]
cc @Julesssss since you graciously volunteered to make sure I didn't break IOUs with this <3

**Comment by stitesExpensify** [comment]
Also if anyone can think of other test cases based on the code this touches, please suggest it!

File: src/libs/actions/Report.js
Comment (Julesssss) [inline src/libs/actions/Report.js:1324]: Possibly out of scope, but I wonder if `fetchIOUReportByID` & `fetchIOUReport` should also be migrated to a new API call. (something like `openPaymentDetailsPage`)

File: src/libs/actions/Report.js
Comment (Julesssss) [inline src/libs/actions/Report.js:315]: Good riddance simplifiedReports 👋

File: src/libs/actions/Report.js
Comment (Julesssss) [inline src/libs/actions/Report.js:499]: I think you can also delete this from the Deprecates API file

Comment (Julesssss) [reply src/libs/actions/Report.js:499]: And do a search for the function, because it's referenced in the Readme.md file

File: src/libs/actions/Report.js
Comment (Julesssss) [inline src/libs/actions/Report.js:572]: I don't have any great suggestions here, but it seems a bit confusing to have separate consts for chat|newChat when that distinction doesn't really matter

File: src/pages/ConciergePage.js
Comment (Julesssss) [inline src/pages/ConciergePage.js:27]: Do we still need to check for this?

**Review by Julesssss — COMMENTED** [review]
Tests nicely so far 👍

File: src/pages/ConciergePage.js
Comment (thienlnam) [reply src/pages/ConciergePage.js:27]: Yeah, not part of your changes but I am curious why it is here (maybe when we link to this page somewhere?)

**Review by thienlnam — COMMENTED** [review]
Changes look solid!

File: src/libs/actions/Report.js
Comment (stitesExpensify) [reply src/libs/actions/Report.js:499]: I removed the actual method, but I think the readme is fine as an example even though it's gone. We're eventually going to run out of commands that are chained, but still want to tell contributors what _not_ to do so I think this is as good as any to leave

File: src/libs/actions/Report.js
Comment (stitesExpensify) [reply src/libs/actions/Report.js:572]: I can add a comment if you think it's helpful, but the reason for that is if newChat is empty we do different things in openReport. So if `chat` were to have a value, we would still try to create the report in openReport

File: src/libs/actions/Report.js
Comment (stitesExpensify) [reply src/libs/actions/Report.js:1324]: Yep I agree. Definitely a refactor for another time though 😄

File: src/pages/ConciergePage.js
Comment (stitesExpensify) [reply src/pages/ConciergePage.js:27]: Yeah it looks like we link directly there from oldDot sometimes. There are at least 2 places it's directly routed/rendered. Must just be a fallback

**Review by thienlnam — DISMISSED** [review]
Thanks!

File: src/libs/actions/Report.js
Comment (stitesExpensify) [reply src/libs/actions/Report.js:572]: Went ahead and added the comment. LMK if you think it needs to be more descriptive or anything :)

**Comment by stitesExpensify** [comment]
@Julesssss can you test sending IOUs on iPhone? I seem to be getting some weird slowness / the iou not showing up when settling manually, but it seems to be acting the same way on `main` so IDK. Just want to make sure nothing is broken

**Comment by stitesExpensify** [comment]
Addressed all other comments. FYI this issue is off hold, and the last PR to close out the LHN refactor! 🎉

**Comment by Julesssss** [comment]
> @Julesssss can you test sending IOUs on iPhone? I seem to be getting some weird slowness / the iou not showing up when settling manually, but it seems to be acting the same way on main so IDK. Just want to make sure nothing is broken

Sorry I don't have any time for this today 😫 Happy to do this next week if it's still required.

**Comment by stitesExpensify** [comment]
Thanks @rushatgabhane, I'll take a look at those tests today

**Comment by stitesExpensify** [comment]
Okay so the test is failing due to an infinite loop. 
- When we are testing, we are assuming the user does not have the report in onyx
- That means that when we hit [this line in HandleReportChanged](https://github.com/Expensify/App/blob/89f24b5748fc6cc9b478f7775b4bf3cbd698c6cf/src/libs/actions/Report.js#L792) we don't have the name of the report, and call openReport
- Unfortunately when you call openReport, it doesn't return the reportName when you are just tagged in a report that you don't already have.
- What is happening is that we call [addComment](https://github.com/Expensify/Web-Expensify/blob/c6d343448399135b52ea4dd81a50ddfc3878f91f/lib/ReportAPI.php#L359), which calls [notifyParticipants](https://github.com/Expensify/Web-Expensify/blob/f9a3d33888576af7a5e8601cef41920309558e19/lib/ReportUtils.php#L1854)
- This [has a condition](https://github.com/Expensify/Web-Expensify/blob/f9a3d33888576af7a5e8601cef41920309558e19/lib/ReportUtils.php#L1867-L1869) that only returns the room name if it is a default room, which is what causes the loop.


So solution time!
1. Remove the room type check in notifyParticipants. 
    - Pros: Minimal reason not to do this (we would need to change how we do the notification [here](https://github.com/Expensify/Web-Expensify/blob/f9a3d33888576af7a5e8601cef41920309558e19/lib/ReportUtils.php#L1949-L1953))
    - Cons: It doesn't really solve the issue of the user not having all of the information about a report
2. Always return the report object when calling openReport:
    - Pros: We will always have all of the report data, and in the future if someone changes the name we will get it
    - Cons: Returning a bunch of data we might not always need
3. Pass a flag to return the full report object sometimes when calling openReport:
    - Pros: Less data transfer
    - Cons: Deciding when to use or not use the param is confusing and complicates the method

**Comment by stitesExpensify** [comment]
Any thoughts of that ^ @Julesssss @roryabraham @thienlnam @rushatgabhane ?

**Comment by stitesExpensify** [comment]
Personally I think `2` is the best solution. Let's just return the simplified report object every time we call openReport

**Comment by stitesExpensify** [comment]
4th solution would be to just remove that check, but then if you get added to a workspace room while you are logged in, your browser will break so.. probably not the best choice

**Comment by thienlnam** [comment]
Yeah, I agree that 2/3 sounds good here though leaning more towards 2 as it's the simplest.  Are there any large keys that end up returned that we don't need? IIRC we removed fetching all the policy members as part of logging in because it was a huge key, but we need all report participants for the report so I'm not sure what other key we would care enough about not returning that it impacted performance

**Comment by roryabraham** [comment]
Agree, was just about to say `2` seems like the best approach to me.

**Comment by stitesExpensify** [comment]
Solution 2 is working with the manual tests but not in the automated tests. Trying to figure out why.

**Comment by stitesExpensify** [comment]
Figured out why. In our unit tests we don't actually call the server (duh). This means that we were never getting the report name from the server when calling openReport so the loop persisted. The solution is to have openReport always have an optimistic reportName which makes sense regardless of this bug. I am going to put this on hold though, because this isn't actually fixing the problem and getting the reportName yet, the web PR does that.

**Comment by stitesExpensify** [comment]
@rushatgabhane were you able to successfully test this PR? I know that it's on hold but I'd like to have it pre-approved so that we can merge as soon as the other PR hits production.

**Comment by JmillsExpensify** [comment]
I'll float in Slack so we can get someone on this.

**Comment by stitesExpensify** [comment]
@Julesssss if you have the bandwidth now, testing would be much appreciated

File: src/libs/actions/Report.js
Comment (Julesssss) [reply src/libs/actions/Report.js:573]: thanks

**Review by Julesssss — DISMISSED** [review]
Testing well for me 👍

**Review by roryabraham — DISMISSED** [review]
According to the [React Native Upgrade Helper](https://react-native-community.github.io/upgrade-helper/?from=0.69.3&to=0.71.0-rc.0), it looks like we shouldn't commit that `contents.xcworkspacedata` file. I don't really know if it matters, but I'd suggest just adding it to the `.gitignore` since it was explicitly removed from the template app at some point.

**Comment by stitesExpensify** [comment]
Nice catch. Remove the file!

**Comment by stitesExpensify** [comment]
Just need @thienlnam for the final review!

**Review by thienlnam — APPROVED** [review]

**Comment by JmillsExpensify** [comment]
Who's taking care of the PR Reviewer checklist?

**Comment by thienlnam** [comment]
<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

File: src/libs/actions/Report.js
Comment (neil-marcellini) [inline src/libs/actions/Report.js:513]: This caused [a regression](https://github.com/Expensify/App/issues/12629). Whenever a report is opened the name is reset to the default, and it's only set back to the proper name, like `#admins`, when the response comes back.

Comment (stitesExpensify) [reply src/libs/actions/Report.js:513]: Thanks for letting me know and taking care of it!

**Pull request merged**
